### PR TITLE
Improve doc

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Update documentation structure and content

* **What is the current behavior?** (You can also link to an open issue here)
Compilation instructions and dependency management is not clear and spread across pages

* **What is the new behavior (if this is a feature change)?**
Simplify documentation to help new users understanding the build chain and use it.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: doc/basics/compilation.md
Comment (MathiasPaulin) [inline doc/basics/compilation.md:15]:
```suggestion
* OpenGL 4.1+ / GLSL 410+
```

File: doc/basics/dependencies.md
Comment (MathiasPaulin) [inline doc/basics/dependencies.md:13]:
```suggestion
Qt, which needs to be installed and passed to cmake through the variable `Qt5_DIR` (see documentation at
```

**Review by MathiasPaulin — COMMENTED** [review]
Seems quite a good improvement in readability. I'll fetch and compile the doc to verify the new structure and content. 
BTW, perhaps you can add somewhere that we encourage users to have dedicated build directory and install directory (the later being manage automatically if the default install dir is used) for each build type (Release, Debug) and that compiling Radium in Debug mode needs to have the dependencies (at least assimp) compiled and installed in Debug mode.

**Comment by nmellado** [comment]
> Seems quite a good improvement in readability. I'll fetch and compile the doc to verify the new structure and content.
> BTW, perhaps you can add somewhere that we encourage users to have dedicated build directory and install directory (the later being manage automatically if the default install dir is used) for each build type (Release, Debug) and that compiling Radium in Debug mode needs to have the dependencies (at least assimp) compiled and installed in Debug mode.

Hum right, this is suggested at multiple places, but not clearly stated. I'll let you do you review run, and we will add that with your other suggestions (if any).

File: doc/basics/compilation.md
Comment (dlyr) [inline doc/basics/compilation.md:155]:
```suggestion
## Integration with Visual Studio (Microsoft Windows)
```

File: doc/main.md
Comment (dlyr) [inline doc/main.md:41]:
```suggestion
 - @ref conceptsmanual : high level concepts used in Radium Engine
```

File: doc/basics/dependencies.md
Comment (dlyr) [inline doc/basics/dependencies.md:12]:
```suggestion
We developed a series of tools to fetch and compile these dependencies easily, except for
```

**Review by dlyr — APPROVED** [review]

**Comment by dlyr** [comment]
Maybe clean the history before merge ? (or squash, but I think two commit may be better)

**Review by MathiasPaulin — APPROVED** [review]

**Comment by MathiasPaulin** [comment]
In the next days (or weeks), I'll propose a new section on "How to write your own plugin" so that even applications that do not support plugins can used the added functionalities (e.g. headless app). I gave an online course to master students, on the spirit of the "ExamplePluginWithLib" test folder, on how to do that and I just have to clean the used repo before submitting it as a PR.

**Comment by nmellado** [comment]
> In the next days (or weeks), I'll propose a new section on "How to write your own plugin" so that even applications that do not support plugins can used the added functionalities (e.g. headless app). I gave an online course to master students, on the spirit of the "ExamplePluginWithLib" test folder, on how to do that and I just have to clean the used repo before submitting it as a PR.

Btw, there is already a page called *how to write your own plugin* in the developer manual

**Comment by MathiasPaulin** [comment]
> Btw, there is already a page called _how to write your own plugin_ in the developer manual

Sure, there is one and it is helpful for developing plugins. 
But I think that this way of extending Radium by developing plugins is very limited as only BaseApplication derived apps will benefit from the functionalities. 
If we want also headless-app (or non Qt-based GUI apps) to benefit from these, we should prefer to develop a library exposing the functionality and interface this library with BaseApplication derived apps by developing a plugin while headless-app will directly link to the library.
This way of extending Radium was the aim of the demo "ExamplePluginWithLib" but, for now, this demo was not updated according to recent PR and does not compile anymore (it is not included in the CI nor in the tests).
There are also other advantages of extending Radium by a lib + plugin, we will discuss about this in a next steering committee.

**Pull request merged**
